From f1dd5aedc82f25be67a306535f3a99f43f618f52 Mon Sep 17 00:00:00 2001
From: Kaushal Billore <kaushal.billore@intel.com>
Date: Fri, 10 Jan 2020 21:39:49 +0530
Subject: [PATCH] [Patch] Pick the USB input device for AOSP camera

This patch help to select the USB audio input device if available

Signed-off-by: Kaushal Billore <kaushal.billore@intel.com>
---
 src/com/android/camera/VideoModule.java | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index e8222f6..6b2efb3 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -47,6 +47,7 @@ import android.provider.MediaStore.Video;
 import android.view.KeyEvent;
 import android.view.View;
 import android.widget.Toast;
+import android.media.AudioDeviceInfo;
 
 import com.android.camera.app.AppController;
 import com.android.camera.app.CameraAppUI;
@@ -1144,7 +1145,19 @@ public class VideoModule extends CameraModule
         mMediaRecorder.setAudioSource(MediaRecorder.AudioSource.CAMCORDER);
         mMediaRecorder.setVideoSource(MediaRecorder.VideoSource.CAMERA);
         mMediaRecorder.setProfile(mProfile);
-        mMediaRecorder.setVideoSize(mProfile.videoFrameWidth, mProfile.videoFrameHeight);
+	{
+		Context mContext = mActivity.getApplicationContext();
+		AudioManager mAudioManager = (AudioManager) mContext.getSystemService(Context.AUDIO_SERVICE);
+		AudioDeviceInfo[] deviceList = mAudioManager.getDevices(AudioManager.GET_DEVICES_INPUTS);
+		for (int index = 0; index < deviceList.length; index++) {
+			Log.d(TAG, "Device : " + deviceList[index] + " deviceId = " + deviceList[index].getId() + " deviceType = " + deviceList[index].getType() + " getProductName = " + deviceList[index].getProductName() + " getAddress " + deviceList[index]. getAddress());
+			if (deviceList[index].getType() == AudioDeviceInfo.TYPE_USB_DEVICE) {
+				mMediaRecorder.setPreferredDevice(deviceList[index]);
+				Log.d(TAG, "mMediaRecorder.setPreferredDevice to Device : " + deviceList[index] + " deviceId = " + deviceList[index].getId() + " deviceType = " + deviceList[index].getType());
+			}
+		}
+	}
+	mMediaRecorder.setVideoSize(mProfile.videoFrameWidth, mProfile.videoFrameHeight);
         mMediaRecorder.setMaxDuration(mMaxVideoDurationInMs);
 
         setRecordLocation();
-- 
2.7.4

